kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -90,6 +93,15 @@ public Protocol getProtocol() {
         return protocol;
     }
 
+    public CreateCheckpointIterator getCreateCheckpointIterator(
+            TableClient tableClient) {
+        return new CreateCheckpointIterator(
+                tableClient,
+                logSegment,
+                System.currentTimeMillis() - 110000000 /* TODO - fix this */
```

Comment:
I will post a separate PR to fix this. In order to get the retention period from the DeltaLog metadata, we need interval parsing code. Will be copying it from the standalone/Spark. Not a small amount of code, that's why putting it in a separate PR.